galaxy_importer/collection.py

```diff
@@ -49,10 +50,17 @@ def import_collection(file, filename=None, logger=None):
 
 
 def _import_collection(file, filename, logger):
-    with tempfile.TemporaryDirectory() as extract_dir:
+    with tarfile.open(fileobj=file, mode='r') as pkg_tar:
+        manifest_file = pkg_tar.extractfile('MANIFEST.json')
+        manifest_data = json.load(manifest_file)
+    metadata = manifest_data['collection_info']
+    sub_path = 'ansible_collections/{}/{}'.format(metadata['namespace'], metadata['name'])
+    file.seek(0)
```

Comment:
Could this extract the archive to tmp with generic sub_path, load the metadata, then rename the sub_path dir.

Probably doesn't matter a ton. Maybe eliminates one re-open. One advantage might that the extract could just be a 'tar' subprocess which might be less picky/finicky than 'tarfile' module.

Just an idea, nothing required.

Reply:
Actually that is very helpful, it also keeps the validation checks in one place.